core/windows_and_tabs/window_snap.py

```diff
@@ -299,14 +325,169 @@ def __init__(self, left, top, right, bottom):
 }
 
 
+def _snap_next(windows: list[Any], target_layout: RelativeScreenPos) -> Optional[int]:
+    for index, window in enumerate(windows):
+        if window is None:
+            return index
+        try:
+            _snap_window_helper(
+                window,
+                target_layout,
+            )
+            return index
+        except Exception as e:
+            print(f"Failed to snap window: {e}")
+    return None
+
+
+def _snap_layout(window_layout: WindowLayout):
+    """Split the screen between multiple windows."""
+    global last_focus_was_done_by_snap_layout
+    global currently_snapping_layout
+    currently_snapping_layout = True
+    import copy
+
+    target_layout = copy.deepcopy(window_layout.layout)
```

Comment:
Is deep copying really needed here?

Reply:
Since we're modifying the underlying data structure I think some level of copying is necessary. Maybe we could get away with the shallow copy? I'm not really a python expert here but I want to make sure that we don't modify the structure that the running process depends on-and I am modifying the underlying array to keep track of where I am in the snapping process.